app/src/main/java/com/ncnf/event/EventBuilder.java

```diff
@@ -32,13 +31,8 @@ public EventBuilder(DatabaseService db){
 
     public CompletableFuture<Event> build(String eventId){
 
-        CompletableFuture<DatabaseResponse> event = db.getData(EVENTS_COLLECTION_KEY + eventId);
-        return event.thenApply(task -> {
-            if (task.isSuccessful()) {
-                return this.build((Map<String, Object>) task.getResult());
-            }
-            return null;
-        });
```

Comment:
There is redundancy with the database/builder/eventBuilder. Should it be merged ?

Reply:
This is deleted.